src/compiler/crystal/macros/methods.cr

```diff
@@ -400,6 +400,24 @@ module Crystal
         interpret_check_args { symbolize }
       when "class_name"
         interpret_check_args { class_name }
+      when "docs"
+        interpret_check_args(named_params: ["comment"]) do
+          is_comment = if named_args && (comment_arg = named_args["comment"]?)
+                         comment_arg.raise "named argument 'comment' to ASTNode#docs must be a BoolLiteral, not #{comment_arg.class_desc}" unless comment_arg.is_a?(BoolLiteral)
+
+                         comment_arg
+                       else
+                         BoolLiteral.new false
+                       end
+
+          docs = self.doc || ""
+
+          if is_comment.value
+            docs = docs.gsub("\n", "\n# ")
```

Comment:
I'm wondering would it be more ergonomic to return all lines commented out... It would be more consistent, that's for one.

```suggestion
            docs = "# #{docs. gsub("\n", "\n# ")}"
```

Reply:
That would most likely make the first line have double `#`. Only new lines need the prefix since the first one is handled by the hard coded `#` in the comment it was called in.

Reply:
That's because you hardcoded it, it could have been `{{ @call.docs(comment: true) }}` (without leading `#`) as well. I'm just thinking that it's bit weird and inconsistent to omit the leading `#`, while the rest of the lines are commented out.

Reply:
I hear your point, @Sija. But the purpose of the `comment` option is specifically for embedding the result in a doc comment.
IMO the ergonomics at the call site with `# {{ node.docs(comment: true) }}` is superior to `{{ node.docs(comment: true)` because it makes the comment function so obvious.

Reply:
I'd be down with a dedicated method for this use case. May also make it more clear that the method is specific to doc generation vs trying to be generic to handle a use case that doesn't event exist yet.

I'll dwell on a name, but `#doc_comment` sounds pretty good to me :+1:

Reply:
Are we thinking this be a new method that returns a `MacroId` with `#` prefixes in addition to, or instead of another `#docs` method that returns the unaltered `StringLiteral`?